core/src/events/mod.rs

```diff
@@ -194,8 +187,8 @@ pub struct EventDetails<T: Config> {
 }
 
 impl<T: Config> EventDetails<T> {
-    // Attempt to dynamically decode a single event from our events input.
-    fn decode_from(
+    /// Attempt to dynamically decode a single event from our events input.
+    pub fn decode_from(
```

Comment:
Perhaps no need for this to be `pub` now that `Events` is our logical "entry point" into this stuff? :)